inductiva/molecules/scenarios/protein_solvation/protein_solvation.py

```diff
@@ -86,28 +105,67 @@ def simulate_async(
         Args:
             simulation_time: The simulation time in ns.
             integrator: The integrator to use for the simulation. Options:
-                - "md" (Molecular Dynamics): Accurate leap-frog algorithm for 
+                - "md" (Molecular Dynamics): Accurate leap-frog algorithm for
                 integrating Newton's equations of motion.
                 - "sd" (Steepest Descent): Stochastic dynamics integrator with
                 leap-frog scheme.
-                - "bd" (Brownian Dynamics): Euler integrator for Brownian or 
-                position Langevin dynamics. 
-                
-            For more details on the integrators, refer to the GROMACS 
-            documentation at 
+                - "bd" (Brownian Dynamics): Euler integrator for Brownian or
+                position Langevin dynamics.
+
+            For more details on the integrators, refer to the GROMACS
+            documentation at
             https://manual.gromacs.org/current/user-guide/mdp-options.html.
 
             nsteps_minim: Number of steps for energy minimization.
         """
+
+        #Compute charge if it is not provided
+        if self.charged is None:
+            self.charged = self.compute_charge()
+
+        #Edit commands.json according to the charge of the protein
+        commands_path = os.path.join(self.template_dir, "commands.json")
+        replace_params_in_template(self.template_dir, "commands.json.jinja", {
```

Comment:
The replacement for this specific parameters should happen if it is charged, right? Otherwise, it is empty.